File: Memento/Java/ConceptualSample/Originator.java
public class Originator {
    //內部狀態
    private String state = "";
    public String getState() {
        return state;
    }
    public void setState(String state) {
        this.state = state;
    }
    //創建一個備忘錄
    public Memento createMemento() {
        return new Memento(this.state);
    }
    //恢復一個備忘錄
    public void restoreMemento(Memento _memento) {
        this.setState(_memento.getState());
    }
}

File: Memento/GoLang/ConceptualSample/originator.go
package main

type Originator struct {
	state string
}

func (o *Originator) CreateMemento() *Memento {
	return &Memento{state: o.state}
}

func (o *Originator) RestoreMemento(m *Memento) {
	o.state = m.state
}

func (o *Originator) SetState(state string) {
	o.state = state
}

func (o *Originator) GetState() string {
	return o.state
}
